txscript/opcode.go

```diff
@@ -875,12 +883,24 @@ func opcodeFalse(op *parsedOpcode, vm *Engine) error {
 // raw data (bytes) to the data stack.
 func opcodePushData(op *parsedOpcode, vm *Engine) error {
 	vm.dstack.PushByteArray(op.data)
+
+	// Used for 2025 upgrade
+	if op.opcode.value >= OP_DATA_1 && op.opcode.value <= OP_DATA_75 {
+		vm.metrics.AddOPCost(int(op.opcode.value))
+	} else if op.opcode.value >= OP_PUSHDATA1 && op.opcode.value <= OP_PUSHDATA4 {
+		vm.metrics.AddOPCost(len(op.data))
+	}
+
 	return nil
 }
 
 // opcode1Negate pushes -1, encoded as a number, to the data stack.
 func opcode1Negate(op *parsedOpcode, vm *Engine) error {
-	vm.dstack.PushInt(scriptNum(-1))
+	vm.dstack.PushInt(*makeScriptNumFromInt64(-1))
+
+	// Used for 2025 upgrade
```

Comment:
Same here, all AppOPCost is 2025 upgrade. We should just add this comment to the AppOPCost method itself.